src/main/scala/mesosphere/marathon/api/v2/AppTasksResource.scala

```diff
@@ -97,11 +99,19 @@ class AppTasksResource @Inject() (
 
     if (scale && wipe) throw new BadRequestException("You cannot use scale and wipe at the same time.")
 
-    if (scale) {
-      val deploymentF = taskKiller.killAndScale(pathId, findToKill, force)
+    if (noOp) {
+      result(groupManager.app(pathId)) match {
+        case None => unknownApp(pathId)
+        case Some(appInfo) =>
+          var tasks = findToKill(taskTracker.appTasksLaunchedSync(pathId))
+          tasks = taskKiller.getTasksToKill(pathId, appInfo, tasks, killCount)
```

Comment:
`findToKill` and `getTasksToKill` should be one function, which is used in dry run and kill mode.